consumer/src/main/java/com/klungerbo/streams/pulsar/ConsumerWorker.java

```diff
@@ -0,0 +1,121 @@
+package com.klungerbo.streams.pulsar;
+
+import com.klungerbo.streams.pulsar.utils.FileUtils;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Partially adapted from https://www.baeldung.com/apache-pulsar
```

Comment:
Code adapted from should be specified at the very top. There is also no explanation of what this class actually is.